loky/backend/_thread_pool_limiters.py

```diff
@@ -0,0 +1,506 @@
+    "openblas": {
+        "set": "openblas_set_num_threads",
+        "get": "openblas_get_num_threads"},
+    "mkl": {
+        "set": "MKL_Set_Num_Threads",
+        "get": "MKL_Get_Max_Threads"},
+}
+
+# Helpers for the doc and test names
+ALL_USER_APIS = set(impl['user_api'] for impl in SUPPORTED_IMPLEMENTATIONS)
+ALL_PREFIXES = [prefix for impl in SUPPORTED_IMPLEMENTATIONS
+                for prefix in impl['filename_prefixes']]
+INTERN_APIS = list(MAP_API_TO_FUNC.keys())
+
+
+class _ThreadPoolLibrariesWrapper:
+    # Wrapper around classic C-libraries for scientific computations which use
+    # thread-pools. Give access to functions to set and get the maximum number
+    # of threads they are allowed to used for inner parallelism.
+
+    cls_thread_locals = threading.local()
+
+    def __init__(self):
+        self._load()
+
+    def get_limit(self, prefix, user_api, limits):
+        if prefix in limits:
+            return limits[prefix]
+        if user_api in limits:
+            return limits[user_api]
+        return None
+
+    def set_thread_limits(self, limits=None, user_api=None):
+        """Limit maximal number of threads used by supported C-libraries.
+
+        Return a list with the modules where the maximal number of threads
+        available has been modified.
+        """
+        if isinstance(limits, int) or limits is None:
+            if user_api in ('all', None):
+                user_api = ALL_USER_APIS
+            elif user_api in ALL_USER_APIS:
+                user_api = (user_api,)
+            else:
+                raise ValueError("user_api must be either 'all', 'blas' or "
+                                 "'openmp'. Got {} instead.".format(user_api))
+            limits = {api: limits for api in user_api}
+            prefixes = []
+        else:
+            if isinstance(limits, list):
+                # This should be a list of module, for compatibility with
+                # the result from get_thread_limits.
+                limits = {module['prefix']: module['n_thread']
+                          for module in limits}
+
+            if not isinstance(limits, dict):
+                raise TypeError("limits must either be an int, a dict or None."
+                                " Got {} instead".format(type(limits)))
+
+            # With a dictionary, can set both specific limit for given modules
+            # and global limit for user_api. Fetch each separately.
+            prefixes = [module for module in limits if module in ALL_PREFIXES]
+            user_api = [module for module in limits if module in ALL_USER_APIS]
+
+        report_threadpool_size = []
+        modules = self._load(prefixes=prefixes, user_api=user_api)
+        for module in modules:
+            n_thread = self.get_limit(module['prefix'], module['user_api'],
+                                      limits)
+            if n_thread is not None:
+                set_func = module['set']
```

Comment:
I find "set" not explicit enough in retrospect.

```python
            if n_thread is not None:
                set_num_threads = module['set_num_threads']
```

Same comment for `"get"` => `"get_num_threads"`.